### DISCUSSION TITLE
idea: Delete images from gallery

### DISCUSSION THREAD

**Discussion by psyafter** [voidlabs/mosaico#625] [Ideas]
Nice to have support option to delete image from gallery (with server callback for deletion)

**Comment by bwl21** [comment]
This might be a bit difficult as it may break existing archived newsletters. It also needs to consider computed resizes.

Perhaps it is better to have an extra image management application, which provides a usasge report. I guess this is beyond the scope of mosaico.

Gallery will be crowded over the time. So it should have some filters.

**Comment by bago** [comment]
The "jquery-file-upload" backend already provides a way to delete files.

It could be easy to add a "trash-can" button over the images in the gallery, but like Bernhard said, maybe this is not something "friendly" because images may be in use.

In our application (embed use-case) we provide integration with MoxieManager and this already gives users the ability to manage/delete images. Other embed solutions may want to use a different gallery for each "edited email" (no "persistence" between templates). Given how easy is to upload an image most people will find it easier to reupload it instead of looking for it in the gallery.

**Comment by DylanHDonsky** [comment]
could we not add to the metadata or somewhere that stores what images are being used in a template? and if you are using a database system it could be a column showing the images used in the template and if they are in use they can't be deleted but if they are then they can go in the trash. i am working on such a feature the issue i am having is finding some variable that stores what images are currently saved in the template even as just path urls.
